src/utils.js

```diff
@@ -209,5 +209,29 @@ go.utils = {
         return today.format('YYYY-MM-DD hh:mm:ss.SSS');
     },
 
+    date_difference: function(date1, date2) {
+        // returns the difference between the dates in days - true adds decimals
+        return moment(date1).diff(moment(date2), 'days', true);
+    },
+
+    fire_returning_user_metrics: function(im, contact) {
+        var today = go.utils.get_today(im.config);
+        contact.extra.last_seen = today;
+        var user_registered = contact.extra.status === 'refugee' || contact.extra.status === 'migrant';
+        var no_last_returning_metric = contact.extra.last_returning_metric_fire === undefined;
+        var old_last_returning_metric = go.utils.date_difference(today, contact.extra.last_returning_metric_fire) > 7;
+
+        if (user_registered && (no_last_returning_metric || old_last_returning_metric)) {
```

Comment:
Please just add a comment to explain what this does. We know now, but in 3 months we might go "eh?"